pkg/controller/servicebindingrequest/annotations/resource_test.go

```diff
@@ -0,0 +1,48 @@
+package annotations
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDiscoverBindingType(t *testing.T) {
+	type args struct {
+		value     string
+		expected  bindingType
+		wantedErr error
+	}
+
+	assertDiscoverBindingType := func(args args) func(t *testing.T) {
+		return func(t *testing.T) {
+			got, err := discoverBindingType(args.value)
+			if args.wantedErr != nil {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, args.expected, got)
+			}
+		}
+	}
+
+	t.Run("env/secret", assertDiscoverBindingType(args{
+		value:    "binding:env:object:secret",
+		expected: BindingTypeEnvVar,
+	}))
+
+	t.Run("env/configmap", assertDiscoverBindingType(args{
+		value:    "binding:env:object:configmap",
+		expected: BindingTypeEnvVar,
+	}))
+
+	t.Run("volumemount/secret", assertDiscoverBindingType(args{
+		value:    "binding:volumemount:secret",
```

Comment:
Shouldn't this be `binding:volumemount:object:secret` ?

Reply:
> Shouldn't this be `binding:volumemount:object:secret` ? 

I couldn't find any mention of `binding:volumemount:object:secret` in the working copy, but could `binding:volumemount:secret`. Can you point out any resource that makes you believe that is the right value?

* https://github.com/redhat-developer/service-binding-operator/search?q=volumemount&unscoped_q=volumemount

Reply:
In the hindsight, I wonder why we used object in the case for env too.. like `servicebindingoperator.redhat.io/status.dbConfigMap-db.port: binding:env:object:configmap`
:thinking:  Anyhow the current implementation in the master is like this and since it is refactoring we can keep it. The annotations, in anycase, would be modified further

Reply:
I couldn't find the `binding:volumemount:object:scret` mention anywhere in the code base.

Yes, amending the annotations syntax will land as a different PR.